File: minegame-shared/src/main/java/top/nipuru/minegame/shared/processor/PlayerInfoUpdateHandler.java
package top.nipuru.minegame.shared.processor;

import lombok.AllArgsConstructor;
import top.nipuru.minegame.common.message.shared.PlayerInfoUpdateNotify;
import top.nipuru.minegame.common.processor.RequestDispatcher;
import top.nipuru.minegame.shared.SharedServer;

@AllArgsConstructor
public class PlayerInfoUpdateHandler implements RequestDispatcher.Handler<PlayerInfoUpdateNotify> {

    private final SharedServer server;

    @Override
    public void handle(RequestDispatcher.ResponseSender responseSender, PlayerInfoUpdateNotify request) throws Exception {
        server.getPlayerInfoManager().insertOrUpdate(request.getPlayerInfo());
    }

    @Override
    public Class<PlayerInfoUpdateNotify> interest() {
        return PlayerInfoUpdateNotify.class;
    }
}

File: minegame-shared/src/main/java/top/nipuru/minegame/shared/processor/PlayerInfoUpdateHandler.kt
package top.nipuru.minegame.shared.processor

import com.alipay.remoting.AsyncContext
import top.nipuru.minegame.common.message.shared.PlayerInfoUpdateNotify
import top.nipuru.minegame.common.processor.RequestDispatcher
import top.nipuru.minegame.shared.player.PlayerInfoManager

class PlayerInfoUpdateHandler : RequestDispatcher.Handler<PlayerInfoUpdateNotify> {

    override fun handle(asyncCtx: AsyncContext, request: PlayerInfoUpdateNotify) {
        PlayerInfoManager.insertOrUpdate(request.playerInfo)
    }

    override fun interest(): Class<PlayerInfoUpdateNotify> {
        return PlayerInfoUpdateNotify::class.java
    }
}
